sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -135,6 +135,7 @@ export function createTokenCycler(
     get shouldRefresh(): boolean {
       return (
         !cycler.isRefreshing &&
+        (token.refreshesOnTimestamp && token.refreshesOnTimestamp < Date.now()) &&
```

Comment:
Not sure if i completely understand what should the exact behavior for refresh on be between `shouldRefresh` and `mustRefresh` APIs.

Reply:
Updated to above behavior